x/ibc-account/keeper/handshake.go

```diff
@@ -77,5 +60,49 @@ func (k Keeper) OnChanOpenTry(
 		return sdkerrors.Wrap(channeltypes.ErrChannelCapabilityNotFound, err.Error())
 	}
 
+	// Register interchain account if it does not already exist
+	_, _ = k.RegisterInterchainAccount(ctx, portID)
+	return nil
+}
+
+func (k Keeper) OnChanOpenAck(
+	ctx sdk.Context,
+	portID,
+	channelID string,
+	counterpartyVersion string,
+) error {
+	k.SetActiveChannel(ctx, portID, channelID)
```

Comment:
I was thinking maybe it would be better to store the full Channel type rather than the id.

Reply:
I think the ID is fine, any reason to store full channel?

Reply:
I think the ID is also fine. Agreed.